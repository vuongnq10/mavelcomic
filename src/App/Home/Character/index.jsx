import T from 'prop-types';
import React from 'react';
import { Link } from 'react-router';
import Card from 'react-bootstrap/Card';

import styles from './styles.css';

const Index = ({ id, name, thumbnail, description }) => (
  <div className={styles.padding}>
    <Card>
      <Card.Body>
        <div className="row">
          <div className="col-lg-3 col-12">
            <Link to={`/hero/${id}`}>
              <img src={`${thumbnail.path}.${thumbnail.extension}`} alt={name} className="img-thumbnail" />
            </Link>
          </div>
          <div className="col-lg-9 col-12">
            <Card.Title>{name}</Card.Title>
            <Card.Text>{description || 'N/A'}</Card.Text>
          </div>
        </div>
      </Card.Body>
    </Card>
  </div>
);

Index.propTypes = {
  id: T.number.isRequired,
  name: T.string.isRequired,
  thumbnail: T.shape({}).isRequired,
  description: T.string.isRequired,
};

export default Index;
