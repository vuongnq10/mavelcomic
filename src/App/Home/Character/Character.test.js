import React from 'react';
import renderer from 'react-test-renderer';

import Character from './index';

it('Renders Character component', () => {
  const character = renderer
    .create(<Character
      id={123}
      name="Comic"
      thumbnail={{
        path: 'path',
        extension: 'extension'
      }}
      description="`description`"
    />)
    .toJSON();

  expect(character).toMatchSnapshot();
});