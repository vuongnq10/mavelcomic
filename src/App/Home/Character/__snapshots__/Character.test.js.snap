// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Character component 1`] = `
<div
  className={undefined}
>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-3 col-12"
        >
          <a
            onClick={[Function]}
            style={Object {}}
          >
            <img
              alt="Comic"
              className="img-thumbnail"
              src="path.extension"
            />
          </a>
        </div>
        <div
          className="col-lg-9 col-12"
        >
          <div
            className="card-title h5"
          >
            Comic
          </div>
          <p
            className="card-text"
          >
            \`description\`
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
